src/object_async/hello_async.cpp

```diff
@@ -165,8 +168,11 @@ struct AsyncHelloWorker : Nan::AsyncWorker
 {
 
     using Base = Nan::AsyncWorker;
+    // Make this class noncopyable
+    AsyncHelloWorker(AsyncHelloWorker const&) = delete;
+    AsyncHelloWorker& operator=(AsyncHelloWorker const&) = delete;
 
-    AsyncHelloWorker(bool louder, std::string const& name,
+    AsyncHelloWorker(bool louder, const std::string* name,
```

Comment:
Okay I see - if you want to guarantee the string's lifetime is bound to the worker's lifetime we should store a string member by value and in the constructor take a string by value and move it into the member. The user can then decide to either move an argument into the worker's constructor or make a copy (but not using move).